pkgs/desktops/deepin/deepin-desktop-schemas/default.nix

```diff
@@ -27,13 +28,36 @@ stdenv.mkDerivation rec {
   ];
 
   postPatch = ''
+    searchHardCodedPaths
+    patchShebangs .
```

Comment:
Wouldn't this patch things with a build time version of bash, etc?

Reply:
Would you please explain it better, please?

Reply:
Well the problem here is actually https://github.com/NixOS/nixpkgs/issues/33956

So what you should do is only patch things that you know won't be installed.
Might not be feasible though if they utilize a million bash scripts in every package.

Reply:
Let me see if I understand: `patchShebangs` should be used only with files that are used for building the package. If the file is installed, it should not be patched this way because it will bring a dependency that may be available only on the build platform. _So in this case, how should it be patched?_

`patchShebans .` is doing the following:
```
patching script interpreter paths in .
./misc/icons/install_to_hicolor.py: interpreter directive changed from "/usr/bin/python3" to "/nix/store/gbwx5sc4w0i0ipnmsa0nsk88svs983vl-python3-3.7.2/bin/python3"
./misc/etc/acpi/powerbtn.sh: interpreter directive changed from "/bin/sh" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/sh"
./misc/etc/grub.d/35_deepin_gfxmode: interpreter directive changed from "/bin/sh" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/sh"
./misc/scripts/dde-system-daemon-power-refresh.sh: interpreter directive changed from "/bin/sh" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/sh"
./debian/rules: interpreter directive changed from "/usr/bin/make -f" to "/nix/store/8n7v99ii7cg694pz0cfch422zvkdqfv4-gnumake-4.2.1/bin/make -f"
./debian/reconfigure-dde-daemon: interpreter directive changed from "/bin/sh" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/sh"
./network/examples/python/main.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/gbwx5sc4w0i0ipnmsa0nsk88svs983vl-python3-3.7.2/bin/python3"
./network/examples/python/gen_dbus_code.sh: interpreter directive changed from "/bin/bash" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/bash"
./network/examples/set_wired_static_ip.sh: interpreter directive changed from " /bin/bash" to "/nix/store/cinw572b38aln37glr0zb8lxwrgaffl4-bash-4.4-p23/bin/bash"
./network/nm_generator/gen_nm_consts.py: interpreter directive changed from "/usr/bin/env python3" to "/nix/store/gbwx5sc4w0i0ipnmsa0nsk88svs983vl-python3-3.7.2/bin/python3"
```
From those it seems that only `35_deepin_gfxmode` is being installed. What should be done with it?

Reply:
Wow they do a lot of stuff for grub https://github.com/linuxdeepin/dde-daemon/commit/42a4b0a021ed372eccca4a4bcf54daac8205f59d

In that case that file isn't even useful on NixOS (In think).

So in general, what you should do is  `patchShebangs absolute/path/to/file/src` but annoyingly it doesn't support multiple arguments https://github.com/NixOS/nixpkgs/issues/57695

So I get by with a `for in` loop and maybe an array if there's a lot of files.

Reply:
Still seeing these references https://gist.github.com/worldofpeace/d11d8f134a4f95d2742ccabe496ddb14

Perhaps a script that finds executables in the `src` and checks if they exist in `out`
could be helpful to investigate quickly.

Reply:
Fixed.